misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormatterUtils.java

```diff
@@ -0,0 +1,230 @@
+                    syntaxKind == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                return null;
+            }
+            return parent;
+        } else if (parentKind == SyntaxKind.FUNCTION_DEFINITION ||
+                parentKind == SyntaxKind.IF_ELSE_STATEMENT ||
+                parentKind == SyntaxKind.ELSE_BLOCK ||
+                parentKind == SyntaxKind.WHILE_STATEMENT ||
+                parentKind == SyntaxKind.CONST_DECLARATION ||
+                parentKind == SyntaxKind.METHOD_DECLARATION ||
+                parentKind == SyntaxKind.TYPE_DEFINITION) {
+            return parent;
+        } else if (syntaxKind == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+            if (parentKind == SyntaxKind.REQUIRED_PARAM ||
+                    parentKind == SyntaxKind.POSITIONAL_ARG ||
+                    parentKind == SyntaxKind.BINARY_EXPRESSION ||
+                    parentKind == SyntaxKind.RETURN_STATEMENT ||
+                    parentKind == SyntaxKind.REMOTE_METHOD_CALL_ACTION ||
+                    parentKind.equals(SyntaxKind.FIELD_ACCESS) ||
+                    (parentKind == SyntaxKind.FUNCTION_CALL && parent.parent() != null &&
+                            parent.parent().kind() == SyntaxKind.ASSIGNMENT_STATEMENT)) {
+                return null;
+            }
+            return getParent(parent, syntaxKind);
+        } else if (syntaxKind.equals(SyntaxKind.STRING_TYPE_DESC) &&
+                parentKind.equals(SyntaxKind.RECORD_FIELD) && parent.parent() != null &&
+                parent.parent().kind().equals(SyntaxKind.RECORD_TYPE_DESC)) {
+            return getParent(parent, syntaxKind);
+        } else if (parentKind == SyntaxKind.SERVICE_DECLARATION ||
+                parentKind == SyntaxKind.BINARY_EXPRESSION) {
+            if (syntaxKind == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
+                return null;
+            }
+            return parent;
+        } else if (parentKind == SyntaxKind.REQUIRED_PARAM) {
+            return null;
+        } else if (parentKind.equals(SyntaxKind.OBJECT_TYPE_DESC)) {
+            if (parent.parent() != null && parent.parent().kind().equals(SyntaxKind.RETURN_TYPE_DESCRIPTOR)) {
+                return parent.parent().parent().parent();
+            } else if (parent.parent() != null && parent.parent().kind().equals(SyntaxKind.TYPE_DEFINITION)) {
+                return getParent(parent, syntaxKind);
+            } else {
+                return parent;
+            }
+        } else if (parent.parent() != null) {
+            return getParent(parent, syntaxKind);
+        } else {
+            return null;
+        }
+    }
+
+    static int getIndentation(Node node, int indentation, FormattingOptions formattingOptions) {
+        if (node == null) {
+            return indentation;
+        }
+        if (node.parent() != null && (node.parent().kind().equals(SyntaxKind.BLOCK_STATEMENT) ||
+                node.parent().kind().equals(SyntaxKind.FUNCTION_BODY_BLOCK) ||
```

Comment:
can extract `node.parent().kind()` to a variable

Reply:
Done.